### PR TITLE
e2e: add search tests

### PR DESCRIPTION
**Pull request by unknwon**
Adds e2e tests for search, which covers:

- Search visibility:private|public
- Can execute search with search operators

Easier to review by commit.

Part of #10068

### CODE REVIEW

**Comment by rvantonder** [comment]
Please cc @sourcegraph/search or me or add us as reviewers so we can see PRs like these. Thank you 🙏

File: dev/e2e/search_test.go [hunk 1]
```diff
@@ -96,4 +97,18 @@ func TestSearch(t *testing.T) {
 			})
 		}
 	})
+
+	t.Run("execute search with search operators", func(t *testing.T) {
```
Comment (rvantonder) [inline dev/e2e/search_test.go:101]: What is meant by `operators` here? Do you mean like `file:.go`... etc? We use `operator` specifically for `or` and `and`, and haven't used it for things like `file:`. I prefer to call these parameters, but at this stage even keyword or field is better than operator. Christina is working on an [RFC](https://docs.google.com/document/d/1SHky6nodPs1w_zRXz24jB2nq5LbMCJl1u7hcLTGHDL8/edit#) to improve consistent terminology here, just want to make sure we are intentional about it in our code/tests as well.

**Comment by unknwon** [comment]
@rvantonder:
> Please cc @sourcegraph/search or me or add us as reviewers so we can see PRs like these. Thank you 🙏

Good call, will do on following PRs!

File: dev/e2e/search_test.go (hunk 1 above)
Comment (unknwon) [reply dev/e2e/search_test.go:101]: I naively copying from old test description :) I'll rename to "parameters".

Comment (unknwon) [reply dev/e2e/search_test.go:101]: Addressed in https://github.com/sourcegraph/sourcegraph/pull/11761/files#diff-0c2d8e2cacf2d6a1fe7892a135a4d443R101.

**Pull request merged**